pkg/overlay/cache.go

```diff
@@ -132,6 +130,10 @@ func (node *NodeDossier) Online() bool {
 		node.Reputation.LastContactSuccess.After(node.Reputation.LastContactFailure)
 }
 
+func (node *NodeDossier) Vetted(maxStats *NodeStats) bool {
```

Comment:
How about have `VettedFor` for the method name and `stats` instead of `maxStats`?

We also need a doc comment - something like:
> VettedFor checks if the node is vetted for the given stats.